api/src/main/java/org/pmiops/workbench/compliance/ComplianceServiceImpl.java

```diff
@@ -84,4 +92,33 @@ public Integer getMoodleId(String email) throws ApiException {
     }
     return response.getBadges();
   }
+
+  @Override
+  public Map<String, BadgeDetails> getUserBadgesByName(String email) throws ApiException {
+    if (!enableMoodleCalls()) {
+      return null;
+    }
+
+    UserBadgeResponse response =
+        moodleApiProvider.get().getMoodleBadge(RESPONSE_FORMAT, getToken(), email);
+    if (response.getException() != null && response.getException().equals(MOODLE_EXCEPTION)) {
+      if (response.getErrorcode().equals(MOODLE_USER_NOT_ALLOWED_ERROR_CODE)) {
+        throw new ApiException(HttpStatus.NOT_FOUND.value(), response.getMessage());
```

Comment:
Please log this at a WARNING level (or at least INFO), so that we can track this down.